Source/OpalGraphics/CGPath.m

```diff
@@ -311,9 +592,87 @@ void CGPathAddArcToPoint(
   CGFloat y1,
   CGFloat x2,
   CGFloat y2,
-  CGFloat r)
+  CGFloat radius)
 {
-  // FIXME:
+  CGFloat dx1, dy1, dx2, dy2;
+  CGFloat l, a1, a2;
+  CGPoint p;
+
+  p = CGPathGetCurrentPoint(path);
+ 
+  dx1 = p.x - x1;
+  dy1 = p.y - y1;
+  
+  l= dx1*dx1 + dy1*dy1;
+  if (l <= 0)

```

Comment:
I agree with your concern; if you think both current versions are suboptimal though, could you suggest a "reference" version so I can reimplement both and keep it uniform?